src/plugins/intel_gna/src/ops/util/util.hpp

```diff
@@ -274,6 +279,115 @@ static bool has_32bit_output(const std::shared_ptr<ngraph::Node>& node) {
+            std::string error = dilation_limit.GetErrorOrEmpty(filter_dilation_height, filter_stride_width);
+            return Cnn2D::AbstractValidator::ValidationSuccesful(raise_exception, error, node->get_friendly_name(), node->get_type_name());
+        };
+        auto filter_shape = conv_ie->input_value(1).get_shape();
+        if (4 == filter_shape.size() && filter_shape.at(2) > 1 && filter_shape.at(3) > 1) {
+            ov::intel_gna::pass::helper::ConvData conv_data;
+            ov::intel_gna::pass::helper::UpdateConvData(conv_ie, conv_data);
+            if (GNAPluginNS::GNAConvolutionLayer::isMappableFrom2DTo1D(
+                conv_data.input_height, conv_data.input_width, conv_data.input_channel_count,
+                conv_data.filter_height, conv_data.filter_width,
+                conv_data.filter_stride_height, conv_data.filter_stride_width)) {
+                return check_dilation(conv_data.filter_dilation_height, conv_data.filter_dilation_width);
+            }
+            const auto cnn2dValidatorPtr = Cnn2D::AbstractValidator::Create(gnaCompileTarget);
+            if (cnn2dValidatorPtr) {
+                return cnn2dValidatorPtr->ValidateCnn2D(node->get_friendly_name(),
+                    conv_data.input_height, conv_data.input_width, conv_data.input_channel_count,
+                    conv_data.filter_height, conv_data.filter_width, conv_data.filter_channel_count,
+                    conv_data.filter_stride_height, conv_data.filter_stride_width,
+                    conv_data.filter_dilation_height, conv_data.filter_dilation_width,
+                    OvGnaTypeIntFromBytes(config.gnaPrecision.size()), raise_exception);
+            }
+        }
+        return check_dilation(conv_ie->get_dilations()[0], conv_ie->get_dilations()[1]);
+    } else if ((std::dynamic_pointer_cast<ngraph::op::ScaleShiftIE>(node) != nullptr) ||
+               (std::dynamic_pointer_cast<ngraph::op::PowerIE>(node) != nullptr) ||
+               (std::dynamic_pointer_cast<ngraph::opset9::MatMul>(node) != nullptr)) {
+        return true;
+    } else if (std::dynamic_pointer_cast<ngraph::op::FullyConnected>(node) != nullptr) {
+        size_t output_batch_size = node->get_output_shape(0)[0];
+        if (output_batch_size > 8) {
+            if (raise_exception) {
+                THROW_GNA_EXCEPTION << "topology with layer: " + node->get_friendly_name() + ", type: " + node->get_type_name() +
+                ", and batch size(" + std::to_string(output_batch_size) + ") not supported";
+            }
+            return false;
+        }
+        return true;
+    } else if (is_eltwise_add(node) || is_eltwise_mul(node)) {
+        return true;
+    } else if (is_pooling(node)) {
+        auto max_pool = std::dynamic_pointer_cast<ngraph::opset7::MaxPool>(node);
+        if (!max_pool)
+            return false;
+        auto kernels = max_pool->get_kernel();
+        if (2 == kernels.size() && kernels.at(0) > 1 && kernels.at(1) > 1) {
```

Comment:
using `at` is overhead here (don't need to check range as `at` does since `size()` is already checked)
```suggestion
        if (2 == kernels.size() && kernels[0]> 1 && kernels[1] > 1) {
```